cognite/powerops/transformations/transformations.py

```diff
@@ -0,0 +1,397 @@
+import abc
+import arrow
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, ClassVar
+
+import numpy as np
+import pandas as pd
+import yaml
+from cognite.client import CogniteClient
+from cognite.client.utils import ms_to_datetime
+from cognite.powerops.utils.cdf.calls import retrieve_range
+
+from pydantic import BaseModel
+
+from logging import getLogger
+
+logger = getLogger(__name__)
+
+
+class RelativeDatapoint(BaseModel):
+    offset_minute: float
+    offset_value: float
+
+def _relative_datapoints_to_series(
+    relative_datapoints: list[RelativeDatapoint], start_time: datetime, shift_minutes: int = 0
+) -> pd.Series:
+    """Converts 'relative datapoints' to a pandas Series
+
+    Example:
+        >>> start_time = datetime(2000, 1, 1, 12)
+        >>> relative_datapoints = {
+        ...     0: 42,
+        ...     60: 420,
+        ...     1440: 4200,
+        ... }
+        >>> _relative_datapoints_to_series(relative_datapoints, start_time)
+        2000-01-01 12:00:00      42
+        2000-01-01 13:00:00     420
+        2000-01-02 12:00:00    4200
+        dtype: int64
+    """
+    datapoint_values = [dp.offset_value for dp in relative_datapoints] # consider creating class method for this
+    return pd.Series(
+        datapoint_values,
+        index=[
+            start_time + timedelta(minutes=int(dp.offset_minute) + shift_minutes) for dp in relative_datapoints
+        ],
+    )
+
+
+class Transformation(BaseModel, abc.ABC):
+    transformation_name: ClassVar[str]
+
+    @classmethod
+    def load(cls, transformation_: dict[str, Any]) -> "Transformation":
```

Comment:
```suggestion
    def load(cls, transformation_: dict[str, Any]) -> Self:
```
`from typing_extensions import Self`